core/dbt/contracts/results.py

```diff
@@ -268,10 +269,36 @@ def from_execution_results(
         )
         return cls(metadata=meta, results=processed_results, elapsed_time=elapsed_time, args=args)
 
+    @classmethod
+    def compatible_previous_versions(self):
+        return [
+            ("run-results", 4),
+        ]
+
+    @classmethod
+    def upgrade_schema_version(cls, data):
+        """This overrides the "upgrade_schema_version" call in VersionedSchema (via
+        ArtifactMixin) to modify the dictionary passed in from earlier versions of the run_results."""
+        run_results_schema_version = get_artifact_schema_version(data)
+        if run_results_schema_version <= 5:
```

Comment:
can we clarify what's special about version 5?

Reply:
Good call - added an inline comment explaining